src/round_constants.rs

```diff
@@ -1,113 +1,32 @@
-            let round_constant = Scalar::from_hash(hasher.clone());
-            round_constants.push(round_constant);
-
-            hasher.reset();
-            hasher.input(round_constant.as_bytes());
-        }
-
-        hasher.reset();
-        hasher.input(b"hades252_partial_rounds");
-
-        // Generate the constants for the partial rounds
-        for _ in 0..(num_partial_rounds * t) {
-            let round_constant = Scalar::from_hash(hasher.clone());
-            round_constants.push(round_constant);
-
-            hasher.reset();
-            hasher.input(round_constant.as_bytes());
-        }
-
-        RoundConstants {
-            constants: round_constants,
-        }
-    }
-}
+lazy_static! {
+  pub static ref ROUND_CONSTANTS: [Scalar; 960] = {
+    let bytes = include_bytes!("ark.bin");
 
-pub struct RoundConstantsIterator<'a> {
-    index: usize,
-    end_index: usize,
-    constants: &'a Vec<Scalar>,
-}
-
-impl<'a> RoundConstantsIterator<'a> {
-    pub fn new(constants: &'a Vec<Scalar>) -> RoundConstantsIterator<'a> {
-        RoundConstantsIterator {
-            index: 0,
-            end_index: constants.len() - 1,
-            constants: constants,
-        }
-    }
-}
-
-impl<'a> Iterator for RoundConstantsIterator<'a> {
-    type Item = &'a Scalar;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.index > self.end_index {
-            return None;
-        }
-
-        let scalar = &self.constants[self.index];
-        self.index = self.index + 1;
-
-        Some(scalar)
-    }
+    unsafe { std::ptr::read(bytes.as_ptr() as *const _) }
+  };
 }
 
 #[cfg(test)]
 mod test {
-    use super::*;
-
-    #[test]
-    fn test_round_constants() {
-        let num_full_rounds = 8;
-        let num_partial_rounds = 127;
-        let t = 6;
-
-        let expected_num_constants = (num_full_rounds + num_partial_rounds) * t;
-
-        let round_constants = RoundConstants::generate(num_full_rounds, num_partial_rounds, t);
-        assert_eq!(expected_num_constants, round_constants.constants.len());
+  use super::*;
 
-        // Check each element is non-zero
-        let non_zero_element = round_constants
-            .constants
-            .iter()
-            .find(|&&x| x == Scalar::zero());
-        assert!(non_zero_element.is_none());
+  #[test]
+  fn test_round_constants() {
+    // Check each element is non-zero
+    let non_zero_element = ROUND_CONSTANTS.iter().find(|&&x| x == Scalar::zero());
+    assert!(non_zero_element.is_none());
 
-        let mut constants_iter = round_constants.iter();
-        for _ in 0..expected_num_constants {
-            let x = constants_iter.next();
+    let mut constants_iter = ROUND_CONSTANTS.iter();
+    for _ in 0..960 {
+      let x = constants_iter.next();
 
-            assert!(x.is_some());
-            assert_ne!(&Scalar::zero(), x.unwrap());
-        }
+      assert!(x.is_some());
+      assert_ne!(&Scalar::zero(), x.unwrap());
```

Comment:
We can let `Scalar::zero()` instead of calling it two times. 
Not important anyways.

Reply:
Since it's a test, I don't mind the verbosity. But I don't think this test make sense as is, so I'm going to refactor it a bit anyway! :)